src/staking/manager.ts

```diff
@@ -0,0 +1,834 @@
+      stakingInput.stakingTimelock,
+    );
+
+    const unbondingPsbt = staking.toUnbondingPsbt(
+      unsignedUnbondingTx,
+      stakingTx,
+    );
+    const signedUnbondingPsbtHex = await this.btcProvider.signPsbt(
+      unbondingPsbt.toHex()
+    );
+    const signedUnbondingTx = Psbt
+      .fromHex(signedUnbondingPsbtHex).extractTransaction();
+
+    // Add covenant unbonding signatures
+    // Convert the params of covenants to buffer
+    const covenantBuffers = params.covenantNoCoordPks.map((covenant) =>
+      Buffer.from(covenant, "hex"),
+    );
+    const witness = createCovenantWitness(
+      // Since unbonding transactions always have a single input and output,
+      // we expect exactly one signature in TaprootScriptSpendSig when the
+      // signing is successful
+      signedUnbondingTx.ins[0].witness,
+      covenantBuffers,
+      covenantUnbondingSignatures,
+      params.covenantQuorum,
+    );
+    // Overwrite the witness to include the covenant unbonding signatures
+    signedUnbondingTx.ins[0].witness = witness;
+
+    return signedUnbondingTx;
+  }
+
+  /**
+   * Creates a signed withdrawal transaction on the unbodning output expiry path 
+   * that is ready to be sent to the BTC network.
+   * @param stakingInput - The staking inputs.
+   * @param stakingParamVersion - The param version that was used to create the
+   * delegation in Babylon chain
+   * @param earlyUnbondingTx - The early unbonding transaction.
+   * @param feeRate - The fee rate in satoshis per byte.
+   * @returns The signed withdrawal transaction.
+   */
+  async createSignedWithdrawEarlyUnbondedBtcTransaction(
+    stakingInput: StakingInputs,
+    stakingParamVersion: number,
+    earlyUnbondingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const params = getBabylonParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!params) {
+      throw new Error(
+        `Unable to find staking params for version ${stakingParamVersion}`,
+      );
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating signed unbonding transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      params,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { psbt: unbondingPsbt } = staking.createWithdrawEarlyUnbondedTransaction(
+      earlyUnbondingTx,
+      feeRate,
+    );
+
+    const signedWithdrawalPsbtHex = await this.btcProvider.signPsbt(
+      unbondingPsbt.toHex()
+    );
+    return Psbt.fromHex(signedWithdrawalPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a signed withdrawal transaction on the staking output expiry path 
+   * that is ready to be sent to the BTC network.
+   * @param stakingInput - The staking inputs.
+   * @param stakingParamVersion - The param version that was used to create the EOI
```

Comment:
Text needs update to be similar to the one in the previous withdrawal method. Same for all below withdrawal methods.